File: packages/cli/test/fixtures/ranking-corpora/typescript/src/handlers/update.ts
import { logError, logInfo } from '../util/logger.js';

/** Changes an existing user's display name. */
export function updateUser(id: number, name: string): void {
  if (id <= 0) {
    logError('updateUser', new Error('id must be positive'));
    return;
  }
  logInfo(`updated user ${name}`);
}

File: packages/cli/test/fixtures/ranking-corpora/go/internal/handlers/update.go
package handlers

import (
	"errors"

	"github.com/lien-fixtures/ranking-go/internal/util"
)

// UpdateUser changes an existing user's display name, logging the outcome
// through the shared util package.
func UpdateUser(id int, name string) error {
	if id <= 0 {
		err := errors.New("id must be positive")
		util.LogError("UpdateUser", err)
		return err
	}
	util.LogInfo("updated user " + name)
	return nil
}
